### ISSUE TITLE
Regarding the use of Real samples from the test set

### ISSUE THREAD

**Issue by wclemonade** [ZSHsh98/NSG-VD#3]
Hello, I would like to ask about the evaluation protocol.
You mentioned that the MSRVTT test set was further split into 9,000 samples for testing and 1,000 samples for validation, and that 100 Real samples from this validation split were used as the reference set for computing distances during testing.

Would this setup introduce any form of label or distribution leakage?
Since the reference set is also drawn from the original test Real samples, the Real samples in the test set will naturally be closer to the Real reference set, which may bias the distance-based classification results.

Or perhaps I misunderstood some part of the procedure — I would appreciate your clarification.

<img width="1392" height="814" alt="Image" src="https://github.com/user-attachments/assets/2695a8ae-e145-4d40-a0c7-b536a5c8f120" />

**Comment by ZSHsh98** [comment]
Thank you for your attention!

I'd like to clarify that our evaluation setup is specifically designed to prevent any data leakage. The key point is that we maintain strict separation between reference sets and evaluation samples throughout our validation and testing phases. Specifically, during validation we use references from the test split to evaluate on the validation split; during testing we use references from the validation split to evaluate on the test split. **In both cases, the reference samples are completely disjoint from the evaluation samples being classified**.

Furthermore, if you'd like to ensure maximum independence in your own experiments, I recommend splitting your target domain dataset into three distinct parts: for example, allocate 100 samples for the reference set, 200 samples for validation, and use the remaining samples for testing. This approach guarantees complete separation between reference, validation, and test sets.

I hope this clarification is helpful. Please feel free to reach out if you have any further questions!

**Comment by wclemonade** [comment]
> Thank you for your attention!
> 
> I'd like to clarify that our evaluation setup is specifically designed to prevent any data leakage. The key point is that we maintain strict separation between reference sets and evaluation samples throughout our validation and testing phases. Specifically, during validation we use references from the test split to evaluate on the validation split; during testing we use references from the validation split to evaluate on the test split. **In both cases, the reference samples are completely disjoint from the evaluation samples being classified**.
> 
> Furthermore, if you'd like to ensure maximum independence in your own experiments, I recommend splitting your target domain dataset into three distinct parts: for example, allocate 100 samples for the reference set, 200 samples for validation, and use the remaining samples for testing. This approach guarantees complete separation between reference, validation, and test sets.
> 
> I hope this clarification is helpful. Please feel free to reach out if you have any further questions!

Hello, thank you for your detailed explanation — it does help. I still have one concern: all of these subsets are ultimately drawn from the MSRVTT dataset. Even if the splits do not overlap, there is still an inherent similarity among samples from the same dataset. Would it not be more appropriate to use the real samples from the training set as the reference set instead?

Additionally, in real-world testing, we usually do not know in advance which samples are real or fake. Would relying on a reference set of real samples implicitly assume prior knowledge about the data distribution? If such information is already known beforehand, wouldn’t this setting be closer to a few-shot scenario rather than a purely unseen test?
I may be misunderstanding something, so I would greatly appreciate any clarification you can provide.

**Comment by wclemonade** [comment]
<img width="739" height="518" alt="Image" src="https://github.com/user-attachments/assets/9f756102-c794-4aff-b495-f81d629dac80" />

When I use MSR-VTT as the reference set, the results are quite satisfactory, as shown above.

However, when I replace the reference set with Kinetics-400 (val), the performance drops significantly.

<img width="797" height="499" alt="Image" src="https://github.com/user-attachments/assets/0dfefa3e-1d22-42b0-a17a-99ec935e48b2" />
From the results above, it appears that the choice of real reference dataset has a substantial impact on performance.
In real-world deployment, I do not know in advance whether a test video is real or fake, and I believe that having prior knowledge of the test distribution is not a reasonable assumption. If such information were available beforehand, the evaluation setting would be closer to a few-shot scenario. I may be misunderstanding something, so I would greatly appreciate any clarification you can provide.

**Comment by ZSHsh98** [comment]
Thank you for your detailed analysis and for sharing your experimental results. You've raised an excellent point about the importance of reference set selection, and your observations are actually consistent with our findings and the core principles of our method.

In our paper, we trained on Kinetics-400 (real) and SEINE/Pika (generated), but used MSR-VTT as the reference set during testing because **domain alignment between reference and test distributions is crucial**. MSR-VTT shares closer characteristics with our test videos compared to Kinetics-400, which primarily contains human action scenes. When you switch to Kinetics-400 as a reference, the domain gap leads to the performance drop you observed.

For practical applications, we recommend:

1. **Domain-Specific Reference Set**: If you know the target domain of your test videos (e.g., social media clips, surveillance footage, face videos), collect 50-100 real videos from that specific domain or from closely related ones as your reference set. The closer the domain match, the better the detection performance. 

2. **Reference Set Size**: Our experiments show that even 50-100 real videos provide stable performance when they're domain-relevant.

3. **Training Data Enhancement**: If you have additional real video data from relevant domains, incorporating them during training will further improve generalization.

You're right that building a universal video detector is extremely challenging. In practice, we could envision NSG-VD as a domain-adaptable system, allowing users to easily swap reference sets according to their specific use cases.

We appreciate you highlighting this important practical consideration! Thank you again for your interest and support!
